GitTfs/Core/GitChangeInfo.cs

```diff
@@ -46,15 +47,72 @@ public struct ChangeType
         public static IEnumerable<GitChangeInfo> GetChangedFiles(TextReader reader)
         {
             string line;
+            List<GitChangeInfo> changes = new List<GitChangeInfo>(); ;
             while (null != (line = GetDiffTreeLine(reader)))
             {
                 var change = Parse(line);
 
                 if (FileMode.GitLink == change.NewMode)
                     continue;
 
-                yield return change;
+                changes.Add(change);
             }
+            var filterd = FilterNotRepresentable(changes);
+            return filterd;
+        }
+
+        private static IEnumerable<GitChangeInfo> FilterNotRepresentable(IEnumerable<GitChangeInfo> changed)
+        {
+            //filter case only renames out, they cannot be represented in TFS
+            var remainingChanges = changed.Where(cha => cha.Status != GitChangeInfo.ChangeType.RENAMEEDIT ||
+                String.Compare(cha.path, cha.pathTo, System.StringComparison.OrdinalIgnoreCase) != 0 ||
+                cha.newSha != cha.oldSha).ToList();
+
+            foreach (var cha in remainingChanges)
+            {
+                if(cha.Status == GitChangeInfo.ChangeType.RENAMEEDIT)
```

Comment:
Do you mean something like FilterAddDeleted and FilterRenameEdit?
Or which operations should I extract?